lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1487,12 +1487,82 @@ void NodeOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                          MLIRContext *context) {
   results.insert<patterns::EmptyNode>(context);
 }
+// A register with constant reset and all connection to either itself or the
+// same constant, must be replaced by the constant.
+struct foldResetMux : public mlir::RewritePattern {
+  foldResetMux(MLIRContext *context)
+      : RewritePattern(RegResetOp::getOperationName(), 0, context) {}
+  LogicalResult matchAndRewrite(Operation *op,
+                                PatternRewriter &rewriter) const override {
+    auto reg = dyn_cast_or_null<RegResetOp>(op);
+    if (!reg)
+      return failure();
+    auto reset = dyn_cast_or_null<ConstantOp>(reg.resetValue().getDefiningOp());
+    auto resetSignal = reg.resetSignal();
+    // Check for a valid const reset.
+    // TODO: Shoud we add a check for resetSignal.isa<BlockArgument>()?
```

Comment:
Why?

Reply:
Actually checked with SFC, not required.